pkg/server/status_local_file_retrieval.go

```diff
@@ -59,6 +62,24 @@ func profileLocal(
 			return nil, err
 		}
 		return &serverpb.JSONResponse{Data: buf.Bytes()}, nil
+	case serverpb.ProfileRequest_GOROUTINE:
+		p := pprof.Lookup("goroutine")
+		if p == nil {
+			return nil, status.Error(codes.InvalidArgument, "unable to find goroutine profile")
```

Comment:
Interesting choice of `InvalidArgument`, I would just stick to internal error for all of these.